app/mm-redux/actions/threads.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ThreadTypes, PostTypes, UserTypes} from '@mm-redux/action_types';
+
+import {Client4} from '@mm-redux/client';
+import {DispatchFunc, GetStateFunc} from '@mm-redux/types/actions';
+import {UserThreadList} from '@mm-redux/types/threads';
+
+import ThreadConstants from '../constants/threads';
+import {logError} from './errors';
+import {forceLogoutIfNecessary} from './helpers';
+
+export function getThreads(userId: string, teamId: string, before = '', after = '', perPage = ThreadConstants.THREADS_CHUNK_SIZE, unread = true) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let userThreadList: undefined | UserThreadList;
+
+        try {
+            userThreadList = await Client4.getUserThreads(userId, teamId, before, after, perPage, true, true, unread);
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        if (userThreadList?.threads?.length) {
+            const flat = require('array.prototype.flat');
+            dispatch({
+                type: UserTypes.RECEIVED_PROFILES_LIST,
+                data: flat(userThreadList.threads.map(({participants: users}) => users)),
+            });
+
+            dispatch({
+                type: PostTypes.RECEIVED_POSTS,
+                data: {posts: userThreadList.threads.map(({post}) => post)},
+            });
+        }
+
+        dispatch({
```

Comment:
I didn't think we'd reach this code in the error case (it will only be undefined if there's an error). 

I've added a check just in case I've missed something.